README.md

```diff
@@ -61,7 +61,15 @@ Run the command to build the code `mvn clean install`
 
 If all goes well, the build should be successful and a war should be produced in `web/web.rya/target/web.rya.war`
 
-Note: To perform a build of the geomesa/lucene indexing, run the build with the profile 'indexing' `mvn clean install -P indexing`
+Note: The following profiles are available to tailor the build:
+
+
+| Profile ID | Purpose |
+| ---------- | ------- |
+| geoindexing | perform a build of the geomesa/lucene indexing |
+| mongodb | build with mongoDB configuration (defaults to accumulo) |
+
+To run the build with the profile 'geoindexing' `mvn clean install -P geoindexing`.
```

Comment:
By product of the rebase.  Initially this branch was clone off of a branch that Andrew was working on (he had made some progress on the external optimizer framework that I picked up).  That pulled in a number of other commits that have been merged -- I didn't know whether you wanted my PR to highlight only my changes or if you wanted other peoples commits included.